### PR TITLE
Fix benefit summary request body keys and conditional option text

### PR DESCRIPTION
**Pull request by raquelromano-gov**
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3723
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3685

Tested this against CI, and the POST successfully downloads a customized BSL.
Now writing unit tests.

This also includes some refactoring, so here is a summary of the changes:
  - move pdf fetch into action creator so that success or failure actions are dispatched; currently we don't do much with these, but they should be handled with an appropriate error message (we may want to include this in https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3780)
  - removed hard-coded military options for usability test
  - maps the BSL options returned by the initial request to the request body options for customizing the PDF
  - renames `optionsToInclude` to `requestOptions`
  - guts our `apiRequest` helper function so we can use `common/helpers/api/commonApiClient` (which should probably happen in a bunch of places throughout the codebase to reduce code duplication)

### CODE REVIEW

File: src/js/letters/reducers/index.js
Comment (annekainicUSDS) [inline src/js/letters/reducers/index.js]: At this point they haven't toggled anything though, correct? This happens before the user toggles anything, and is used as a default list of the options that apply to this user that can then be rendered in the UI.

**Review by annekainicUSDS — COMMENTED** [review]
This is a great start. This type of conditional logic is really hard to handle in a clear, straightforward way. I'm finding the `getBenefitOptionText` helper a little hard to reason through (the inconsistencies with checking just if `currentValue` exists vs. also checking if `currentValue !== 'unavailable'`. It's hard to handle this cleanly because the data we're getting isn't clean or consistent. But what if we had this data structured more similarly to what Rachael has in the ticket? Such as: 
```
const benefitOptionText = {
  hasServiceConnectedDisabilities: {
    true: {
      veteran: 'You have one or more service-connected disabilities.',
      dependent: undefined
    },
    false: {
      veteran: 'You do not have one or more service-connected disabilities.',
      dependent: undefined
    },
  },
  ...
}

getBenefitOptionText(currentOption, currentValue, userType) {
  return benefitOptionText[currentOption][currentValue][userType];
}
```
Then if what `getBenefitOptionText` returns is undefined we know not to display anything. Not sure if this will work or not, and we'll have to handle the `currentValue` differently if it's an actual value as opposed to `true` or `false` (can probably pass this in `getBenefitOptionText(currentOption, currentValue && currentValue !== false, userType)` or something like that). What do you think?

File: src/js/letters/utils/helpers.js
Comment (raquelromano-gov) [inline src/js/letters/utils/helpers.js]: OK, will take a look.

**Comment by raquelromano-gov** [comment]
I like the idea of restructuring the function to get the display text! Will make things much more readable.  I'll give that a shot later today.

File: src/js/letters/reducers/index.js
Comment (raquelromano-gov) [inline src/js/letters/reducers/index.js]: Yeah, this should say "all possible displayed options that the user may toggle on/off"

File: src/js/letters/actions/letters.js
Comment (raquelromano-gov) [inline src/js/letters/actions/letters.js]: We should decide what to do about failed PDF downloads and add to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3780

File: src/js/letters/utils/helpers.js
Comment (raquelromano-gov) [inline src/js/letters/utils/helpers.js]: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3685 has some updates (bolding and language), but still has one detail to be resolved.

File: src/js/letters/actions/letters.js
Comment (annekainicUSDS) [inline src/js/letters/actions/letters.js]: Yes, great point. Probably a question for @rroueche and @phiden. I assume display some kind of error message to the user?

File: src/js/letters/utils/helpers.js
Comment (annekainicUSDS) [inline src/js/letters/utils/helpers.js]: Minor typo: `conditionall` should be `conditionally`

File: src/js/letters/utils/helpers.js
Comment (annekainicUSDS) [inline src/js/letters/utils/helpers.js]: Why the break here?

File: src/js/letters/utils/helpers.js
Comment (annekainicUSDS) [inline src/js/letters/utils/helpers.js]: I don't think this one is right. From what Ian told me (and we might want to triple check this), `awardEffectiveDate` shouldn't appear as a clickable option in the list because it's not something they can optionally include in their letter or not. Therefore it doesn't map to anything that we send in the request for the pdf for bsl.

File: src/js/letters/utils/helpers.js
Comment (annekainicUSDS) [inline src/js/letters/utils/helpers.js]: Not sure we have to worry about `awardEffectiveDate` (see comment below), let's confirm with @rroueche

**Review by annekainicUSDS — APPROVED** [review]
Looks great! Tested locally well too (only thing was because `awardEffectiveDate` and `monthlyAward` are being mapped to the same thing, they both toggle when one is clicked, but I think we can remove `awardEffectiveDate` anyway, something to confirm with @rroueche). Awesome PR!

File: src/js/letters/utils/helpers.js
Comment (raquelromano-gov) [inline src/js/letters/utils/helpers.js]: Ah, OK I did not know that--thanks. I will remove it and double-check with Rachael tomorrow.

I also plan to add more unit test to the components before merging, but will try to merge this tomorrow.

Comment (raquelromano-gov) [inline src/js/letters/utils/helpers.js]: Actually, the request body option `monthlyAward` for generating the BSL seems to simultaneously control whether both of these appear in the PDF (the award amount and the award effective date). I've asked in #evss channel if this is by design, and if this is what is done in eBenefits.

**Comment by raquelromano-gov** [comment]
Another edge case that I think should be considered in a separate PR (maybe once we figure out the `awardEffectiveDate` issue: if all options are unselected, the letter looks like this:
image

Maybe if EVSS changes the letter generator service to always show the award effective date (as opposed to coupling it with the award amount), then that date can always be shown in the "VA Benefit Information" section

File: test/letters/containers/VeteranBenefitSummaryLetter.unit.spec.jsx
Comment (raquelromano-gov) [inline test/letters/containers/VeteranBenefitSummaryLetter.unit.spec.jsx]: This is all more brittle than I would like, even though I added an ID to each table to make it simpler. I think I've spent too much time on this test already, so wondering what you think about making any testing improvements in a separate PR.

File: src/js/letters/actions/letters.js
Comment (phiden) [inline src/js/letters/actions/letters.js]: do we have a global error message for failed downloads from blue button or similar we could use?

Comment (annekainicUSDS) [inline src/js/letters/actions/letters.js]: I have no idea. Perhaps you could ask someone on the Kudos team?

Comment (phiden) [inline src/js/letters/actions/letters.js]: asking!

**Review by annekainicUSDS — APPROVED** [review]
Looks good! I'm having trouble testing this locally because I seem to be having some trouble with id.me. The only consideration is last time I tested this, because `awardEffectiveDate` and `monthlyAward` are mapped to the same field, they both toggle when you check or uncheck one of them. Not a big deal temporarily if we're eventually changing how `awardEffectiveDate` is going to work, but something we'll definitely have to revisit at some point. Perhaps add a TODO so we remember to get back to that?

**Comment by raquelromano-gov** [comment]
Yeah idme is misbehaving for me too--I dropped a note to Tanel in the #identity channel.

For now, I think the coupling of those 2 checkboxes is the best way to go given what Ian has said about what the PDF generator service. I tried to get him to explain in more detail how it works on eBen right now, but it was hard since we had multiple threads.

I think the best thing to do is to get this on staging, show Rachael and explain the behavior of that service, compare to eBen, make a product decision, and go back to EVSS with any requests.

**Pull request merged**
